pkg/volume/cephfs/cephfs_test.go

```diff
@@ -215,13 +215,63 @@ func TestGetSecretNameAndNamespaceForPV(t *testing.T) {
 			expectedNs:    "default",
 			expectedError: nil,
 		},
+		{
+			name:      "pod volume source no secretRef",
+			defaultNs: "default",
+			spec: &volume.Spec{
+				Volume: &v1.Volume{
+					VolumeSource: v1.VolumeSource{
+						CephFS: &v1.CephFSVolumeSource{
+							Monitors: []string{"a", "b"},
+							User:     "user",
+						},
+					},
+				},
+			},
+			expectedName:  "",
+			expectedNs:    "",
+			expectedError: nil,
+		},
+		{
+			name:          "no volume",
+			spec:          &volume.Spec{},
+			expectedName:  "",
+			expectedNs:    "",
+			expectedError: fmt.Errorf("Spec does not reference an CephFS volume type"),
+		},
 	}
 	for _, testcase := range tests {
 		resultName, resultNs, err := getSecretNameAndNamespace(testcase.spec, testcase.defaultNs)
-		if err != testcase.expectedError || resultName != testcase.expectedName || resultNs != testcase.expectedNs {
-			t.Errorf("%s failed: expected err=%v ns=%q name=%q, got %v/%q/%q", testcase.name, testcase.expectedError, testcase.expectedNs, testcase.expectedName,
-				err, resultNs, resultName)
+		switch testcase.name {
+		case "no volume":
+			if err.Error() != testcase.expectedError.Error() || resultName != testcase.expectedName || resultNs != testcase.expectedNs {
+				t.Errorf("%s failed: expected err=%v ns=%q name=%q, got %v/%q/%q", testcase.name, testcase.expectedError, testcase.expectedNs, testcase.expectedName,
+					err, resultNs, resultName)
+			}
+		default:
+			if err != testcase.expectedError || resultName != testcase.expectedName || resultNs != testcase.expectedNs {
+				t.Errorf("%s failed: expected err=%v ns=%q name=%q, got %v/%q/%q", testcase.name, testcase.expectedError, testcase.expectedNs, testcase.expectedName,
+					err, resultNs, resultName)
+			}
 		}
 	}
+}
+
+func TestGetAccessModes(t *testing.T) {
+	tmpDir, err := utiltesting.MkTmpdir("cephfs_test")
+	if err != nil {
+		t.Fatalf("error creating temp dir: %v", err)
+	}
+	defer os.RemoveAll(tmpDir)
 
+	plugMgr := volume.VolumePluginMgr{}
+	plugMgr.InitPlugins(ProbeVolumePlugins(), nil /* prober */, volumetest.NewFakeVolumeHost(tmpDir, nil, nil))
+
+	plug, err := plugMgr.FindPersistentPluginByName("kubernetes.io/cephfs")
+	if err != nil {
+		t.Errorf("Can't find the plugin by name")
+	}
+	if !volumetest.ContainsAccessMode(plug.GetAccessModes(), v1.ReadWriteOnce) || !volumetest.ContainsAccessMode(plug.GetAccessModes(), v1.ReadOnlyMany) || !volumetest.ContainsAccessMode(plug.GetAccessModes(), v1.ReadWriteMany) {
```

Comment:
can you just compare the array directly?

Reply:
ok,thanks,i have fixed it!